vars/provisionNodes.groovy

```diff
@@ -120,10 +120,14 @@ Map call(Map config = [:]) {
         }
     }
 
+    if (!fileExists("${WORKSPACE}/ci")) {
+        // Required when running from a scripted pipeline
+        checkoutScm(cleanAfterCheckout: false)
```

Comment:
But what happens when this code is running in a non-scripted pipeline, as it would if somebody were to build a historic commit prior to all of this work landing?  Or were to be calling this from a non-scripted stage in some other pipeline?

Should this perhaps be moved into the scripted stage directly?  Or do you need to wrap this function with an implementation for scripted stages?

Reply:
Thanks, I've moved it to getFunctionalTestStage.